<a href="https://colab.research.google.com/github/Pogayo/Machine-Learning/blob/master/E_learning_improve.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

The purpose of this notebook is to investigate how students feel about the e-learning program. I will analyze the data from the e-learning satisfaction survey, specifically the question on how the program should be improved?

**Methodology**
1. Will use sentiment analysis to gauge the polarity of the comments --Will then compare the polarity with the responses on how likely the students are to recommend the program to others.
2. I will also extract the specific courses that students had a problem wi

In [2]:
from google.colab import files
uploaded=files.upload()

Saving satisfaction.csv to satisfaction.csv


In [0]:
from textblob import TextBlob
import numpy as np
import pandas as pd
import io
import six    #will be used to check if a variable is a string
import matplotlib.pyplot as plt


In [0]:
raw_data=pd.read_csv(io.BytesIO(uploaded['satisfaction.csv']), header=[1])

In [7]:
raw_data.head()

Unnamed: 0  ... Would you explain in your own words what need to be improved in the training to increase your satisfaction level?
0         NaN  ...  what I may suggest in front office agents depa...                                                                
1         NaN  ...  For me including professional pactices like op...                                                                
2         NaN  ...                                                NaN                                                                
3         NaN  ...                                                NaN                                                                
4         NaN  ...                                                NaN                                                                

[5 rows x 41 columns]

In [0]:
improved=raw_data['Would you explain in your own words what need to be improved \ufeffin the training to increase your satisfaction level?']
improved=improved.drop_duplicates()
improved=improved.dropna()
improved=improved.rename("improve_col")
improved.describe()

In [0]:
polarity=[]
subjectivity=[]
improved_sentiments=pd.DataFrame( data=[], columns=["comment", "polarity", "subjectivity"])

In [0]:
sum_polarity=0
min_improved_polarity=20
max_improved_polarity=-2

for comment in improved:
   blob=TextBlob(comment)
   pol=blob.polarity
   polarity.append(pol)
   sum_polarity+=pol
   if pol>max_improved_polarity:
     max_improved_polarity=pol
   if pol<min_improved_polarity:
     min_improved_polarity=pol

   df=pd.DataFrame({"comment":[comment], "polarity":[pol], "subjectivity":[blob.subjectivity]})
   improved_sentiments=improved_sentiments.append(df, ignore_index=True)
   subjectivity.append(blob.subjectivity)


In [0]:
avg_improved_polarity=sum_polarity/318

In [134]:
improved_sentiments


comment  polarity  subjectivity
0    what I may suggest in front office agents depa...  0.000000        0.0000
1    For me including professional pactices like op...  0.336667        0.4600
2                                       I WAS improved  0.000000        0.0000
3                  I want to increase my English level  0.000000        0.0000
4         Offering uniforms to E learners if possible.  0.000000        1.0000
..                                                 ...       ...           ...
313                     we make increasing in practice  0.000000        0.0000
314  yes their  purpose have it all ready  gainning...  0.200000        0.5000
315  Thank you very much,in this training you have ...  0.300000        0.4500
316  will be  much better if you change place and y...  0.187500        0.4375
317  The training can be extended by having enough ...  0.000000        0.5000

[318 rows x 3 columns]

In [105]:
print("Average Polarity:"+str(avg_improved_polarity))
print("Maximum Polarity:"+str(max_improved_polarity))
print("Minimum Polarity:"+str(min_improved_polarity))

Average Polarity:0.1791349021058141
Maximum Polarity:1.0
Minimum Polarity:-0.30000000000000004


In [138]:
improved_sentiments.sort_values("polarity", inplace=True)
improved_sentiments.head(10)

comment  polarity  subjectivity
87   Please find an IT person (because the machines... -0.300000      0.400000
229  Reducing the mark for sartifite at least it ca... -0.300000      0.400000
288  I would like tell to vatel 3things  increase t... -0.300000      0.400000
71   To help the students to get the transport  bec... -0.291667      0.541667
6    I was working hard and ask  somethings which i... -0.233333      0.545833
225  I would like to ask vatel training to help tra... -0.133333      0.600000
78   to have other training that can help me to be ... -0.125000      0.375000
289  Increasing level of my work , competence with ... -0.125000      0.375000
158  One month of training is not enough, and time ... -0.121875      0.575000
167  How long it takes when I say this I mean that ... -0.120833      0.529167

In [141]:
print("Bottom 10 Comments")
for index, row in improved_sentiments.head(10).iterrows():
     # access data using column names
     print(index, row['comment'])

Bottom 10 Comments
87 Please find an IT person (because the machines was so slow)
229 Reducing the mark for sartifite at least it can be at 65
288 I would like tell to vatel 3things  increase time at least 5h perday  visit hotel for practical's  I thank for this oppportunity
71 To help the students to get the transport  because sometimes it hard for them because we are jobless
6 I was working hard and ask  somethings which is not full understanding ,And working  together with my partners .
225 I would like to ask vatel training to help trainees to get internship and possible job.  also the marks considered for getting certificate may be decreased below to 80 and be like 65 above.
78 to have other training that can help me to be satisfaction
289 Increasing level of my work , competence with other people and to be professionalism in my company
158 One month of training is not enough, and time of practice was very little l.
167 How long it takes when I say this I mean that one month s not

In [110]:
1.179/2*10
0.179/1*10


1.79

# How to improve E-learning findings 
1. Average Polarity is **0.1791349021058141**
2. The polarity is **5.89/10**
3. Positivity of the comments out of **1.79/10**

**Top 10 comments (Positive)**



**Bottom 10 comments(Negative)**

1. Please find an IT person (because the machines was so slow)

2.  Reducing the mark for sartifite at least it can be at 65

3. I would like tell to vatel 3things  increase time at least 5h perday  visit hotel for practical's  I thank for this oppportunity

4. To help the students to get the transport  because sometimes it hard for them because we are jobless

5. I would like to ask vatel training to help trainees to get internship and possible job.  also the marks considered for getting certificate may be decreased below to 80 and be like 65 above.

6. to have other training that can help me to be satisfaction

7. Increasing level of my work , competence with other people and to be professionalism in my company

8. One month of training is not enough, and time of practice was very little l.

9. How long it takes when I say this I mean that one month s not enou

**Most mentioned courses**

**Time**

Time was meantioned 
